packages/@divvi/mobile/src/recaptcha/RecaptchaService.ts

```diff
@@ -0,0 +1,102 @@
+import {
+  Recaptcha,
+  RecaptchaAction,
+  type RecaptchaClient,
+} from '@google-cloud/recaptcha-enterprise-react-native'
+import { Platform } from 'react-native'
+import { getAppConfig } from 'src/appConfig'
+import Logger from 'src/utils/Logger'
+
+const TAG = 'recaptcha/RecaptchaService'
+const RECAPTCHA_ACTION_TIMEOUT_MS = 10000 // 10 seconds
+
+export enum RecaptchaActionType {
+  PHONE_VERIFICATION = 'PHONE_VERIFICATION',
+  KEYLESS_BACKUP = 'KEYLESS_BACKUP',
+}
+
+class RecaptchaServiceImpl {
+  private client: RecaptchaClient | null = null
+  private initializationPromise: Promise<void> | null = null
+
+  async initializeIfNecessary(): Promise<void> {
+    if (this.client) return
+    if (this.initializationPromise) return this.initializationPromise
+
+    this.initializationPromise = this.init()
+    try {
+      await this.initializationPromise
+    } catch (err) {
+      // If initialization fails, clear the promise so that we can try again later
+      this.initializationPromise = null
+      throw err
+    }
+  }
+
+  private async init(): Promise<void> {
+    // If client is already initialized, no need to initialize again
+    if (this.client) {
+      return
+    }
+
+    Logger.info(TAG, 'Initializing reCAPTCHA client')
+    const clientKey = this.getSiteKey()
+    if (!clientKey) {
+      Logger.info(TAG, 'No client key found, reCAPTCHA not enabled')
+      return
+    }
+
+    try {
+      this.client = await Recaptcha.fetchClient(clientKey)
+      Logger.info(TAG, 'reCAPTCHA client initialized successfully')
+    } catch (error) {
+      Logger.error(TAG, 'Failed to initialize reCAPTCHA client', error)
+      throw error
+    }
+  }
+
+  /**
+   * Get a reCAPTCHA token for a specific action
+   * @param actionType - The type of action being performed
+   * @returns Promise<string> - The reCAPTCHA token
+   */
+  async getToken(action: RecaptchaActionType): Promise<string | null> {
+    if (!this.isEnabled()) {
+      return null
+    }
+
+    try {
+      await this.initializeIfNecessary()
+      if (!this.client) {
```

Comment:
you could make the "initialize promise" resolve to the client.  then you'd do something like `const client = await this.getClient()` (instead of `await initializeIfNecessary()`) and remove the `!this.client` check.  fwiw, this was the pattern i expected it took a me a second to understand the current init code and checks.